### PR TITLE
Update to 1.17.1

### PR DESCRIPTION
**Pull request by NoComment1105**
Here is the base update to 1.17.1.
There are still 3 project errors that i can't figure out myself.
I'm opening this PR as a draft for now until it's ready for merge
*i didn't know which branch to pr too so i pr'd to master*

### CODE REVIEW

File: src/main/java/me/lambdaurora/keystrokes/gui/KeystrokesColorConfigScreen.java [hunk 1]
```diff
@@ -119,9 +119,9 @@ public void apply_rainbow(boolean newValue)
     {
         this.mod.config.setRainbowText(newValue);
 
-        this.foreground_buttons.forEach(button -> button.active = !newValue);
+        this.foregroundButtons.forEach(button -> button.setActive(true));
         if (this.rainbowSaturationButton != null)
-            this.rainbowSaturationButton.active = newValue;
+            this.rainbowSaturationButton.setActive(true);
```
Comment (NoComment1105) [inline src/main/java/me/lambdaurora/keystrokes/gui/KeystrokesColorConfigScreen.java]: I forgot to mention that his is a bit concern, idk if its right

Comment (LambdAurora) [reply src/main/java/me/lambdaurora/keystrokes/gui/KeystrokesColorConfigScreen.java]: should be setActive(newValue)

Comment (NoComment1105) [reply src/main/java/me/lambdaurora/keystrokes/gui/KeystrokesColorConfigScreen.java]: Pushed a new commit fixing it

**Comment by NoComment1105** [comment]
My genius is imeasureable and my day is ruined

**Pull request closed without merging**